src/conventions/h-group/clue-interpretation/connecting-cards.js

```diff
@@ -149,40 +151,122 @@ function find_unknown_connecting(game, giver, target, reacting, identity, connec
 				return;
 			}
-			return { type: 'finesse', reacting, card: finesse, identities: [identity] };
+			return { type: 'finesse', reacting, card: finesse, bluff, identities: [identity] };
 		}
 		// Finessed card is delayed playable
 		else if (game.level >= LEVEL.INTERMEDIATE_FINESSES && state.play_stacks[finesse.suitIndex] + 1 === finesse.rank) {
-			// Could be duplicated in giver's hand - disallow hidden finesse
-			if (state.hands[giver].some(c => c.clued && game.players[giver].thoughts[c.order].inferred.has(identity))) {
+			// Could be duplicated in giver's hand - disallow hidden finesse unless it could be a bluff.
+			if (!bluff && state.hands[giver].some(c => c.clued && game.players[giver].thoughts[c.order].inferred.has(identity))) {
 				logger.warn(`disallowed hidden finesse on ${logCard(finesse)} ${finesse.order}, true ${logCard(identity)} could be duplicated in giver's hand`);
 				return;
 			}
 
-			return { type: 'finesse', reacting, card: finesse, hidden: true, identities: [finesse.raw()] };
+			return { type: 'finesse', reacting, card: finesse, hidden: true, bluff, identities: [finesse.raw()] };
 		}
 	}
 }
 
+/**
+ * Determines whether a bluff connection is a valid bluff, and updates the connection accordingly.
+ * @param {Game} game
+ * @param {number} giver 				The player index that gave the clue. They cannot deduce unknown information about their own hand.
+ * @param {number} target 				The player index receiving the clue. They will not find self-prompts or self-finesses.
+ * @param {Connection[]} connections	The complete connections leading to the play of a card.
+ * @param {ActualCard[]} promised		The non-hidden play(s) which are promised.
+ * @param {Identity} focusIdentity		The expected identity of the focus.
+ * @returns {Connection[]}
+ */
+export function resolve_bluff(game, giver, target, connections, promised, focusIdentity) {
+	if (connections.length == 0 || !connections[0].bluff)
+		return connections;
+	const { state } = game;
+	const bluffCard = connections[0].card;
+	let bluff_fail_reason = undefined;
+	// Determine the next play if this is a bluff.
+	const next_play = connections.findIndex(connection => connection.card.order == promised[0].order) + 1;
+	// A bluff must be followed only by prompts as otherwise it would not have been a valid bluff target.
+	if (!connections.every((conn, index) => index < next_play || (!conn.hidden && conn.type !== 'finesse')))
+		bluff_fail_reason = `requires additional finesses`;
+
+	if (!bluff_fail_reason) {
+		// A bluff must be recognizable. As such, there should be no connection
+		// between the bluffed card and at least one of the following plays
+		// as known by the player who would play next after the bluff play.
+		let known_bluff = false;
+		const next_player = state.hands.findIndex(hand => hand.find(c => c.order == promised[1].order));
+		const bluff_identities = bluffCard.suitIndex == -1 ?
```

Comment:
This is probably okay currently, but in the future I want to leave open the possibility of adding partial information to cards as they become known, so I would prefer something like `bluffCard.identity() === undefined` here.